File: src/main/java/ru/mrgrd56/api/exception/ConflictException.java
package ru.mrgrd56.api.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.server.ResponseStatusException;

public class ConflictException extends ResponseStatusException {
    public ConflictException() {
        super(HttpStatus.CONFLICT);
    }

    public ConflictException(String reason) {
        super(HttpStatus.CONFLICT, reason);
    }

    public ConflictException(String reason, Throwable cause) {
        super(HttpStatus.CONFLICT, reason, cause);
    }
}

File: src/main/kotlin/ru/mrgrd56/api/exception/ConflictException.kt
package ru.mrgrd56.api.exception

import org.springframework.http.HttpStatus
import org.springframework.web.server.ResponseStatusException

class ConflictException : ResponseStatusException {
    constructor() : super(HttpStatus.CONFLICT)
    constructor(reason: String?) : super(HttpStatus.CONFLICT, reason)
    constructor(reason: String?, cause: Throwable?) : super(HttpStatus.CONFLICT, reason, cause)
}
